krux_sqs/sqs.py

```diff
@@ -0,0 +1,166 @@
+
+    if not logger:
+        logger = get_logger(name=NAME)
+
+    if not stats:
+        stats = get_stats(prefix=NAME)
+
+    boto = Boto3(
+        log_level=args.boto_log_level,
+        access_key=args.boto_access_key,
+        secret_key=args.boto_secret_key,
+        region=args.boto_region,
+        logger=logger,
+        stats=stats,
+    )
+    return Sqs(
+        boto=boto,
+        logger=logger,
+        stats=stats,
+    )
+
+
+def add_sqs_cli_arguments(parser, include_boto_arguments=True):
+    """
+    Utility function for adding SQS specific CLI arguments.
+    """
+    if include_boto_arguments:
+        # GOTCHA: Since many modules use krux_boto, the krux_boto's CLI arguments can be included twice,
+        # causing an error. This creates a way to circumvent that.
+
+        # Add all the boto arguments
+        add_boto_cli_arguments(parser)
+
+    # Add those specific to the application
+    group = get_group(parser, NAME)
+
+
+class Sqs(object):
+    """
+    A manager to handle all SQS related functions.
+    Each instance is locked to a connection to a designated region (self.boto.cli_region).
+    """
+    MAX_RECEIVE_MESSAGES_NUM = 10
+    MAX_RECEIVE_MESSAGES_WAIT = 10
+
+    def __init__(
+        self,
+        boto,
+        logger=None,
+        stats=None,
+    ):
+        # Private variables, not to be used outside this module
+        self._name = NAME
+        self._logger = logger or get_logger(self._name)
+        self._stats = stats or get_stats(prefix=self._name)
+
+        if not isinstance(boto, Boto3):
+            raise NotImplementedError('Currently krux_boto.sqs.Sqs only supports krux_boto.boto.Boto3')
+
+        self._resource = boto.resource('sqs')
+        self._queues = {}
+
+    def _get_queue(self, queue_name):
+        """
+        Returns a queue with the given name.
+        The queue is fetched on the first call (lazy) and cached.
+
+        :param queue_name: :py:class:`str` Name of the queue to get.
+        """
+        if queue_name not in self._queues or self._queues[queue_name] is None:
+            self._queues[queue_name] = self._resource.get_queue_by_name(QueueName=queue_name)
+
+        return self._queues[queue_name]
+
+    def get_messages(self, queue_name):
+        """
+        Returns a list of messages in the given queue.
+
+        :param queue_name: :py:class:`str` Name of the queue to get messages from.
+        """
+        # GOTCHA: Note that not all messages may be returned
+        # http://boto3.readthedocs.org/en/latest/reference/services/sqs.html#SQS.Queue.receive_messages
+        raw_messages = self._get_queue(queue_name).receive_messages(
+            MaxNumberOfMessages=self.MAX_RECEIVE_MESSAGES_NUM,
+            WaitTimeSeconds=self.MAX_RECEIVE_MESSAGES_WAIT
+        )
+
+        result = []
+        for msg in raw_messages:
+            # Parse the strings as JSON so that we can deal with them easier
+            body_dict = simplejson.loads(msg.body)
```

Comment:
If it is intended for the caller to catch JSON parsing errors, you should document that in the docstring. Otherwise you should catch them here, even if you re-throw as a custom exception type.